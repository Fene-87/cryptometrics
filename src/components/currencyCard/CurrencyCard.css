* {
  margin: 0;
  padding: 0;
}

.card {
  background-color: #fb5193;
  width: 100%;
  border: 1px solid #d3d3d3;
  height: 150px;
  padding: 10px 5px;
}

.card-image {
  width: 70px;
  height: 70px;
}

.currency-name {
  color: #fff;
  font-family: 'lato', sans-serif;
}

.price {
  color: #fff;
  font-family: 'Roboto Slab', sans-serif;
  font-size: 12px;
  margin-top: 10px;
}

.price-figures {
  color: #febb02;
  background-color: #ce4277;
  margin-left: 5px;
  padding: 5px 2px;
  border-radius: 5px;
}
