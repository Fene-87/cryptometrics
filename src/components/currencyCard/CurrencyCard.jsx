import React from 'react';

const CurrencyCard = ({
  id,
  name,
  image,
  current_price,
  last_updated,
}) => {
  return (
    <section>
      <div>
        <img src={image} alt="image" />
      </div>
      <h2>{name}</h2>
      <p>{current_price}</p>
    </section>
  );
};

export default CurrencyCard;
