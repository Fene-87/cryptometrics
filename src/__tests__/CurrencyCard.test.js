import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import CurrencyCard from '../components/currencyCard/CurrencyCard';

describe('Currency Card Component', () => {
  test('renders CryptoMetrics on homepage', () => {
    render(
      <BrowserRouter><CurrencyCard /></BrowserRouter>,
    );

    const price = screen.getByText('Current Price:');
    expect(price).toBeInTheDocument();
  });

  test('component match with the snapshot', () => {
    // Assert
    const tree = renderer.create(<BrowserRouter><CurrencyCard /></BrowserRouter>).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
