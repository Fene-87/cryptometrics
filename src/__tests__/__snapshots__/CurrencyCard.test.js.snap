// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Currency Card Component component match with the snapshot 1`] = `
<section
  className="card"
>
  <div>
    <img
      className="card-image"
    />
  </div>
  <h2
    className="currency-name"
  />
  <p
    className="price"
  >
    Current Price: 
    <span
      className="price-figures"
    >
      $
    </span>
  </p>
</section>
`;

exports[`Currency component component match with the snapshot 1`] = `
  <section
    className="card"
  >
    <div>
        <img src={image} alt={name} className="card-image" />
    </div>
    <h2
      className="currency-name"
    >
    </h2>
    <p
      className="price"
    >
      Current Price: 
      <span
       className="price-figures"
      >
        $
      </span>
    </p>
  </section>
`;
